test/e2e/testsuites/azdiskschedulerextender_pod_scheduling_on_failover.go

```diff
@@ -86,5 +84,6 @@ func (t *AzDiskSchedulerExtenderPodSchedulingOnFailover) Run(client clientset.In
 	time.Sleep(30 * time.Second)
 
 	ginkgo.By("checking that the pod for statefulset is running")
-	tStatefulSet.WaitForPodReady()
+	err = tStatefulSet.WaitForPodReady()
```

Comment:
Consider adding a `WaitForPodReadyOrFail()` function that raises the failure. It would be clearer from the stack trace then that the timeout occured during the wait rather than somewhere in the `Run` function.